pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/SlingResourceIncludeProcessor.java

```diff
@@ -77,7 +77,8 @@ public void process(Document document, PreprocessorReader reader, String target,
             log.warn("Could not find include for {}", target);
         }
 
-        reader.push_include(content, target, target, 1, attributes);
+        String includedFilePath = includeResource != null ? includeResource.getPath() : target;
+        reader.push_include(content, includedFilePath, includedFilePath, 1, attributes);
```

Comment:
Nice find! At first I was confused because I had tried testing this with this document structure:
```
master.adoc
a/level2.adoc
a/b/level3.adoc
```
...and that worked perfectly in both master and your branch.
Then I added 2 more level just to be sure:
```
/a/b/c/level4.adoc
/a/b/c/d/level5.adoc
```
...and lo and behold, I saw the breakage in master, and the fix in this branch! So that was quite a surprise, but good find and good fix. I'm not sure I'm following why this fixes it though - would you care to elaborate?

Reply:
It doesn't work by itself. The adjustment you made to "fix" the target on line 43 completes the puzzle. It seems that the `getFile` and `getDir` methods from asciidoctorj don't return the right data every time (only at the top 2 levels it seems). They behave correctly however if when doing the include we tell asciidoctor the exact location of these files (hence why I just get the full path of the resource and send it with the include).

Reply:
Wow, that's quite strange. But, ok! I'll chalk it up as an asciidoctor eccentricity. Thanks for elaborating.